types/puppeteer/index.d.ts

```diff
@@ -2117,10 +2107,10 @@ export interface CoverageEntry {
 }
 
 /** Attaches Puppeteer to an existing Chromium instance */
-export function connect(options?: ConnectOptions): Promise<Browser>;
+export function connect(options?: ConnectOptions & BrowserOptions): Promise<Browser>;
 /** The default flags that Chromium will be launched with */
-export function defaultArgs(options?: DefaultArgsOptions): string[];
+export function defaultArgs(options?: ChromeArgOptions): string[];
 /** Path where Puppeteer expects to find bundled Chromium */
 export function executablePath(): string;
 /** The method launches a browser instance with given arguments. The browser will be closed when the parent node.js process is closed. */
-export function launch(options?: LaunchOptions): Promise<Browser>;
+export function launch(options?: LaunchOptions & ChromeArgOptions & BrowserOptions): Promise<Browser>;
```

Comment:
Shouldn't launch options extend those 2 instead of doing type unions here?
Same for `ConnectOptions` btw.